utils/include/tlv.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Write data to storage.
+ * 
+ * @param ctx Storage context.
+ * @param offset Location where to place the date.
+ * @param data Data to save.
+ * @param data_size Number of bytes to save.
+ * 
+ */
+typedef int (*tlv_storage_write)(void *ctx, uint32_t offset, uint8_t *data, uint32_t data_size);
+
+/**
+ * @brief Read data from storage.
+ * 
+ * @param ctx Storage context.
+ * @param offset Location where to find the date.
+ * @param data Buffor where data will be placed.
+ * @param data_size Number of bytes to read.
+ * 
+ */
+typedef int (*tlv_storage_read)(void *ctx, uint32_t offset, uint8_t *data, uint32_t data_cap);
+
+typedef struct tlv_ctx {
+	struct storage_impl {
+		void *ctx; // Context of the storage API
+		tlv_storage_write write; // Function to write data to storage
+		tlv_storage_read read; // Function to read data from storage
+	} storage; // API to access storage location.
+	uint32_t first_entry_offset; // First valid address where data can be placed.
```

Comment:
If it's an address why it's `uint32_t`? It should be a pointer type, right? or the comment is misleading?

Reply:
In case of Flash storage, we also use an address name, but it is not a pointer. I think the comment could be misleading.

Removed the comment.